src/Host-SDK/Temporal.Worker.Workflows/WorkflowsApi.cs

```diff
@@ -0,0 +1,309 @@
+        Task<PayloadsCollection> ExecuteAsync(string activityName, PayloadsCollection activityArguments, CancellationToken cancelToken);
+        Task<PayloadsCollection> ExecuteAsync(string activityName, PayloadsCollection activityArguments, IActivityInvocationConfiguration invocationConfig);
+        Task<PayloadsCollection> ExecuteAsync(string activityName, PayloadsCollection activityArguments, CancellationToken cancelToken, IActivityInvocationConfiguration invocationConfig);
+
+        Task ExecuteAsync(string activityName);
+        Task ExecuteAsync(string activityName, CancellationToken cancelToken);
+        Task ExecuteAsync(string activityName, IActivityInvocationConfiguration invocationConfig);
+        Task ExecuteAsync(string activityName, CancellationToken cancelToken, IActivityInvocationConfiguration invocationConfig);
+
+        Task ExecuteAsync<TArg>(string activityName, TArg activityArguments) where TArg : IDataValue;
+        Task ExecuteAsync<TArg>(string activityName, TArg activityArguments, CancellationToken cancelToken) where TArg : IDataValue;
+        Task ExecuteAsync<TArg>(string activityName, TArg activityArguments, IActivityInvocationConfiguration invocationConfig) where TArg : IDataValue;
+        Task ExecuteAsync<TArg>(string activityName, TArg activityArguments, CancellationToken cancelToken, IActivityInvocationConfiguration invocationConfig) where TArg : IDataValue;
+
+        Task<TResult> ExecuteAsync<TResult>(string activityName) where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TResult>(string activityName, CancellationToken cancelToken) where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TResult>(string activityName, IActivityInvocationConfiguration invocationConfig) where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TResult>(string activityName, CancellationToken cancelToken, IActivityInvocationConfiguration invocationConfig) where TResult : IDataValue;
+
+        Task<TResult> ExecuteAsync<TArg, TResult>(string activityName, TArg activityArguments) where TArg : IDataValue where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TArg, TResult>(string activityName, TArg activityArguments, CancellationToken cancelToken) where TArg : IDataValue where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TArg, TResult>(string activityName, TArg activityArguments, IActivityInvocationConfiguration invocationConfig) where TArg : IDataValue where TResult : IDataValue;
+        Task<TResult> ExecuteAsync<TArg, TResult>(string activityName, TArg activityArguments, CancellationToken cancelToken, IActivityInvocationConfiguration invocationConfig) where TArg : IDataValue where TResult : IDataValue;
+    }
+
+    // ----------- -----------
+
+    public interface IDeterministicApi
+    {
+        DateTime DateTimeUtcNow { get; }
+
+        Random CreateNewRandom();
+        Guid CreateNewGuid();
```

Comment:
Is this the same as `uuid`?

Reply:
The term for UUID in the Microsoft world is GUID.